### ISSUE TITLE
imagenet example does not work with nccl

### ISSUE THREAD

**Issue by kiukchung** [pytorch/elastic#64]
## 🐛 Bug
TLDR; image net example crashes with nccl
See https://github.com/pytorch/elastic/issues/61

Component (check all that applies):
 * [ ] `state api`
 * [ ] `train_step api`
 * [ ] `train_loop`
 * [ ] `rendezvous`
 * [ ] `checkpoint`
 * [ ] `rollback`
 * [ ] `metrics`
 * [ ] `petctl`
 * [x] `examples`
 * [ ] `docker`
 * [ ] other
 

## To Reproduce

Run the example with `backend=="nccl"`

## Expected behavior
Should run

## Environment
Does not matter

## Additional context
Two potential fixes:

1. Create the `state_tensor` and `state_size` tensor (https://github.com/pytorch/elastic/blob/master/torchelastic/p2p/coordinator_p2p.py#L153) as a gpu tensors if the backend is "nccl"

2. Create a secondary gloo-based process group to sync state from max_rank.

**Comment by kiukchung** [comment]
obsolete as of 0.2.0
